Various fixes and additions.

Flaksim opened the pull request:
*Added Squad Waypoint Marker to ODST Prowler.
*Fixed placement of some items on the bridge.
*Added ODST Backpacks.
*Added ODST Backpacks to vendor.
*Implemented all code changes XO made in PR: #555 (to prevent this merge messing his up)

Flaksim committed:
Various fixes and additions.

*Added Squad Waypoint Marker to ODST Prowler.
*Fixed placement of some items on the bridge.
*Added ODST Backpacks.
*Added ODST Backpacks to vendor.
*Implemented all code changes XO made in PR: #555 (to prevent this merge messing his up)

Outstanding problem: The ODST backpacks, whilst visible on the ground, and working as a backpack, are invisible when equipped, I do not know why.
The icon states for worn can be found in back.dmi and odst.dmi Someone with more knowledge should look at this issue and fix it.

DoctorOctopusTheOctopi commented:
What was changed in the .dmi files?

DoctorOctopusTheOctopi commented on code/__defines/items_clothing.dm:
```diff
@@ -1,6 +1,8 @@
-#define SHOES_SLOWDOWN          -1.0  // How much shoes slow you down by default. Negative values speed you up.
+#define BACKPACK_SLOWDOWN_MOD 0.05 //The modifer to apply to the w_class of an item when calculating slowdown.
+
+#define SHOES_SLOWDOWN         -1.0  // How much shoes slow you down by default. Negative values speed you up.
```
This file can be removed, as not having in this PR won't touch or mess up XO's PR.

DoctorOctopusTheOctopi commented on code/game/objects/items/weapons/storage/storage.dm:
```diff
@@ -26,11 +26,28 @@
+		slowdown_total -= min_slowdown
+
+	if(slowdown_general != slowdown_total)
+		slowdown_general = slowdown_total
```
This file can be removed, as not having in this PR won't touch or mess up XO's PR.

DoctorOctopusTheOctopi commented on code/game/objects/items/weapons/storage/backpack.dm:
```diff
@@ -358,3 +359,52 @@
+	icon = ODST_OVERRIDE
+	name = "Odst Backpack Squad Leader"
+	icon_state = "Odst Backpack Squad Leader"
+	item_state = "Odst L"
```
Replace the spaces in the .dmi with _ and the item state isn't properly set, it has to be the same as what the icon is called in the .dmi

DoctorOctopusTheOctopi commented on code/game/objects/items/weapons/storage/backpack.dm:
```diff
@@ -358,3 +359,52 @@
+	icon = ODST_OVERRIDE
+	name = "Odst Backpack Engineer"
+	icon_state = "Odst Backpack Engineer"
+	item_state = "Odst E"
```
Replace the spaces in the .dmi with _ and the item state isn't properly set, it has to be the same as what the icon is called in the .dmi

DoctorOctopusTheOctopi commented on code/game/objects/items/weapons/storage/backpack.dm:
```diff
@@ -358,3 +359,52 @@
+	icon = ODST_OVERRIDE
+	name = "Odst Backpack Sharpshooter"
+	icon_state = "Odst Backpack Sharpshooter"
+	item_state = "Odst S"
```
Replace the spaces in the .dmi with _ and the item state isn't properly set, it has to be the same as what the icon is called in the .dmi

DoctorOctopusTheOctopi commented on code/game/objects/items/weapons/storage/backpack.dm:
```diff
@@ -358,3 +359,52 @@
+	icon = ODST_OVERRIDE
+	name = "Odst Backpack Medic"
+	icon_state = "Odst Backpack Medic"
+	item_state = "Odst M"
```
Replace the spaces in the .dmi with _ and the item state isn't properly set, it has to be the same as what the icon is called in the .dmi

DoctorOctopusTheOctopi commented on code/game/objects/items/weapons/storage/backpack.dm:
```diff
@@ -358,3 +359,52 @@
+	icon = ODST_OVERRIDE
+	name = "Odst Backpack"
+	icon_state = "Odst Backpack"
+	item_state = "Odst B"
```
Replace the spaces in the .dmi with _ and the item state isn't properly set, it has to be the same as what the icon is called in the .dmi

DoctorOctopusTheOctopi reviewed (dismissed):
The icons also lack a south sprite, my suggestion for naming them would be odst_b for regular backpack, odst_e for engineer backpack, odst_m for medic backpack, odst_s for sharpshooter backpack and odst_sl for squad leader backpack. 

This is what they have to be in the .dmi file and in the item state, for example if you rename the squad leader's backpack to odst_sl the item state would be item_state = "odst_sl" Ping me in discord and we can talk about this more. 😄 

The icons also only have to be added to one file, and can be removed from either back.dmi or odst.dmi I recommend removing them from back.dmi

Flaksim committed:
Made alterations as octo suggested.

The alterations that octo suggested have been implemented, backpacks are still invisible.

Flaksim committed:
Making backpacks work

fixes ODST backpacks, minor tweaks to ODST.

DoctorOctopusTheOctopi reviewed (approved):
This PR looks good to merge then with the discord discussion we've had. Nice job! 👍 🐙
